databricks/koalas/frame.py

```diff
@@ -497,9 +497,7 @@ def count(self):
 
         Returns
         -------
-        Series or DataFrame
-            For each column/row the number of non-NA/null entries.
-            If `level` is specified returns a `DataFrame`.
+        pandas.Series
```

Comment:
should we return a ks.Series here?

Reply:
I had the same question before. We picked pandas Series at that time, but we can rethink now. https://github.com/databricks/koalas/pull/57#discussion_r271578421
cc @thunterdb

Reply:
We should return a pandas series in my opinion. Here is the reasoning:
 - the series is guaranteed to be 'small'
 - it is the output of an aggregation. if it was spark, we would return a Row object
 - koalas series are supposed to be for 'big' data and have different performance expectations when doing a point lookup. It should be instant and not trigger a spark job.

Reply:
If we return a `pd.Series` we need to cast it again to `ks.Series` in case we want to use Koalas methods.
Since the result is guaranteed to be small, we can just return `ks.Series` and I think there is no big impact.

Reply:
@garawalid  there is still an impact in the sense that accessing any of the fields is going to trigger a spark job (> 0.1 sec), let alone the cost of displaying the info, or doing things like iterating. From the code bases I have looked at (seaborn, pandas-profiling) these sections are used with regular python loops and would be very inefficient.

What do you mean 'in case we want to use the koalas methods'? All the functions in the `koalas.*` namespaces should eventually accept both koalas or pandas series / dataframes, because they follow the same specs. Similarly, all the methods of the classes should have the same behaviour.

Reply:
@thunterdb 
I agree about Spark overhead. 

What I meant by koalas methods, when we return `pd.Series` and we apply `filter()`, we will use `pd.Series.filter()`, in this case, there is no problem. But let's assume that we have an other method in Koalas `m()` that doesn't exist in `pd.Series`, users have to cast the result to `ks.Series`  in order to use `m()`

```python
res = df.count()
res = pd2ks(res) # casting 
res.m()
```

Reply:
@garawalid I agree that if we add other methods that are not in the pandas api, then they will not be available. In general so far, we have gone to great lengths to reproduce the pandas API, and there is debate about even adding other functions. This is why for the time being, and because the scenario you are mentioning is hypothetical, I am comfortable returning a pandas object. We can always change that later with no change to the user if we see the benefits outweigh costs in performance. Also, theres is always the possibility of explicitly converting the pandas series to a koalas series.

Reply:
@thunterdb Now, I am convinced that `pd.Series` is a better choice in this case.